lib/whitehall_importer/create_migration.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "gds_api/whitehall_export"
+
+module WhitehallImporter
+  class CreateMigration
+    attr_reader :organisation_content_id, :document_type
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(organisation_content_id, document_type)
+      @organisation_content_id = organisation_content_id
+      @document_type = document_type
+    end
+
+    def call
+      whitehall_migration = ActiveRecord::Base.transaction do
+        record = WhitehallMigration.create!(organisation_content_id: organisation_content_id,
+                                            document_type: document_type)
+
+        whitehall_export = GdsApi.whitehall_export.document_list(organisation_content_id, document_type)
+
+        whitehall_export.each do |page|
+          page["documents"].each do |document|
+            WhitehallMigration::DocumentImport.create!(
+              whitehall_document_id: document["document_id"],
+              whitehall_migration_id: record.id,
+              state: "pending",
+            )
```

Comment:
Sorry I know you didn't introduce this.

It seems a bit unusual that this creates this association without adding it to the record.
For idiomatic Rails I'd have thought we wanted something like:

```
record.document_import << WhitehallMigration::DocumentImport.create!

# though I think we can do it shorter with: 

record.create_document_import!(whitehall_document_id: document["document_id"], state: "pending")
```

Reply:
`record.document_imports.create!` worked for me 👍